src/bottleneck.rs

```diff
@@ -0,0 +1,276 @@
+use crate::common::*;
+
+/// Contains the mapping of each prefix to its bottleneck asn.
+#[derive(Debug, PartialEq)]
+pub(crate) struct Bottleneck {
+    prefix_asn: HashMap<Address, u32>,
+}
+
+impl Bottleneck {
+    pub(crate) fn new() -> Self {
+        Bottleneck {
+            prefix_asn: HashMap::new(),
+        }
+    }
+
+    /// Creates a new `Bottleneck`, reads and parses mrt files, locates prefix and asn bottleneck
+    pub(crate) fn locate(dump: &[PathBuf]) -> Result<Self> {
+        let mut mrt_hm = HashMap::new();
+
+        for path in dump {
+            let buffer = BufReader::new(File::open(path).map_err(|io_error| Error::IoError {
+                io_error,
+                path: path.into(),
+            })?);
+
+            let mut decoder = GzDecoder::new(buffer);
+            Self::parse_mrt(&mut decoder, &mut mrt_hm)?;
+        }
+
+        let mut bottleneck = Bottleneck {
+            prefix_asn: HashMap::new(),
+        };
+        bottleneck.find_as_bottleneck(&mut mrt_hm)?;
+
+        Ok(bottleneck)
+    }
+
+    /// Creates a mapping between a prefix and all of its asn paths, gets the common asns from
+    /// those paths, and considers the last asn (the asn farthest from the originating hop) from
+    /// the common asns to be the bottleneck.
+    fn find_as_bottleneck(
+        &mut self,
+        mrt_hm: &mut HashMap<Address, HashSet<Vec<u32>>>,
+    ) -> Result<(), Error> {
+        let mut prefix_to_common_suffix: HashMap<Address, Vec<u32>> = HashMap::new();
```

Comment:
Need to say somewhere that, in the vector, the first element would be the final AS (so the actual AS of the ip, not some AS on the path), as inseted in find_common_suffix.
And the last element would be the critical AS on the path that matters.